internal/app/tfsec/scanner/scanner.go

```diff
@@ -106,28 +110,43 @@ func (scanner *Scanner) checkRangeIgnored(id string, r block.Range, b block.Rang
 		}
 
 		if strings.Contains(lines[number], ignoreAll) || strings.Contains(lines[number], ignoreCode) {
-			return true
+			foundValidIgnore = true
+			lineValidIgnoreFound = number
+			break
 		}
 	}
 
 	// check the line above the actual resource block
 	if b.StartLine-1 > 0 {
 		line := lines[b.StartLine-1]
-		if ignored := checkLineForIgnore(line, ignoreAll, ignoreCode); ignored {
-			return true
+		if strings.Contains(line, ignoreAll) || strings.Contains(line, ignoreCode) {
+			foundValidIgnore = true
+			lineValidIgnoreFound = b.StartLine-1
 		}
 	}
 
-	return false
-}
-
-func checkLineForIgnore(line, ignoreAll, ignoreCode string) bool {
-	line = strings.TrimSpace(strings.ReplaceAll(strings.ReplaceAll(line, "//", ""), "#", ""))
-	segments := strings.Split(line, " ")
-	for _, segment := range segments {
-		if segment == ignoreAll || segment == ignoreCode {
-			return true
+	if foundValidIgnore {
+		lineWithPotentialExp := lines[lineValidIgnoreFound]
+		if indexExpFound := strings.Index(lineWithPotentialExp, "exp:"); indexExpFound > 0 {
+			debug.Log("Expiration date found on ignore '%s'", lineWithPotentialExp)
+			expDate := lineWithPotentialExp[indexExpFound:]
+			parsedDate, err := time.Parse("exp:2006-01-02", expDate)
+
+			if err != nil {
+				// if we can't parse the date then we don't want to ignore the range
+				debug.Log("Unable to parse exp date in ignore: '%s'. The date format is invalid. Supported format 'exp:yyyy-mm-dd'.", lineWithPotentialExp)
+				return false
+			}
+
+			currentTime := time.Now()
+			ignoreExpirationDateNotBreached := !currentTime.After(parsedDate)
+			if ignoreExpirationDateNotBreached {
+				debug.Log("Ignore ignored - expiration date not breached")
```

Comment:
This logic should be reversed - if the ignore has expired then log to say `Ignore expired, check will be performed Filename: /file/path.tf:lineNo` The required info can be taken from `b`

Reply:
done example in the comment below.